src/main/java/org/kainos/ea/db/AuthDao.java

```diff
@@ -0,0 +1,58 @@
+package org.kainos.ea.db;
+
+import org.apache.commons.lang3.time.DateUtils;
+import org.kainos.ea.cli.Login;
+
+import java.sql.*;
+import java.util.Date;
+import java.util.UUID;
+
+public class AuthDao {
+    private DatabaseConnector databaseConnector;
+
+    public AuthDao(DatabaseConnector databaseConnector) {
+        if(databaseConnector == null)
+        {
+            throw new NullPointerException("Database connector passed in ctor is null");
+        }
+        this.databaseConnector = databaseConnector;
+    }
+
+    public boolean validLogin(Login login) {
+        try (Connection c = databaseConnector.getConnection()) {
+            Statement st = c.createStatement();
+
+            ResultSet rs = st.executeQuery("SELECT Password FROM `User` WHERE Email = '"
+                    + login.getEmail() + "'");
+
+            while (rs.next()) {
+                return rs.getString("password").equals(login.getPassword());
+            }
+
+        } catch (SQLException e) {
+            System.err.println(e.getMessage());
+        }
+
+        return false;
+    }
+
+    public String generateToken(String username) throws SQLException {
+        String token = UUID.randomUUID().toString();
+        Date expiry = DateUtils.addHours(new Date(), 24);
+
+        Connection c = databaseConnector.getConnection();
+
+        String insertStatement = "INSERT INTO Token (Email, Token, Expiry) " +
```

Comment:
Do you need to store the token in the database? When accepting the token you should be able to verify it and use the expiry date on it to authenticate

Reply:
we say no?